src/uct/ib/dc/base/dc_ep.c

```diff
@@ -263,7 +260,12 @@ ucs_status_t uct_dc_ep_check_fc(uct_dc_iface_t *iface, uct_dc_ep_t *ep)
             if (status != UCS_OK) {
                 return status;
             }
+            ucs_assert(!(ep->fc.flags & UCT_DC_EP_FC_FLAG_WAIT_FOR_GRANT));
             ep->fc.flags |= UCT_DC_EP_FC_FLAG_WAIT_FOR_GRANT;
+            /* Update win to avoid sending another fc request if the message,
+             * which invoked this check had not been sent (and therefore FC win
+             * was not decreased)*/
+            UCT_RC_UPDATE_FC_WND(&iface->super, &ep->fc);
```

Comment:
We can check, but we will have to update the win anyway if the grant has been sent already. And in this case we will not be able to use the assertion to keep tracking that no dup grant request is sent.